test/e2e/gameserver_test.go

```diff
@@ -211,6 +213,34 @@ func TestGameServerUnhealthyAfterDeletingPod(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestGameServerUnhealthyAfterReadyCrash(t *testing.T) {
+	t.Parallel()
+
+	l := logrus.WithField("test", "TestGameServerUnhealthyAfterReadyCrash")
+
+	gs := defaultGameServer()
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(defaultNs, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
```

Comment:
any particular reason this one uses `t.Fatalf(...)` while below you use `assert.NoError(...)`

Reply:
Mostly - because copy paste :smile: it's what we have everywhere..

Lower down, it would be super rare to fail (how could you not write to a buffer that doesn't track if it gets received or not?)